src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -567,8 +567,8 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
         List<ProgrammingExerciseStudentParticipation> exportedStudentParticipations = new ArrayList<>();
         for (StudentParticipation studentParticipation : programmingExercise.getStudentParticipations()) {
             ProgrammingExerciseStudentParticipation programmingStudentParticipation = (ProgrammingExerciseStudentParticipation) studentParticipation;
-            if (repositoryExportOptions.isExportAllStudents() || (programmingStudentParticipation.getRepositoryUrl() != null && studentParticipation.getStudent() != null
-                    && studentList.contains(studentParticipation.getStudent().getLogin()))) {
+            if (repositoryExportOptions.isExportAllStudents() || (programmingStudentParticipation.getRepositoryUrl() != null && studentParticipation.getParticipant() != null
+                    && studentParticipation.getStudents().stream().map(User::getLogin).anyMatch(studentList::contains))) {
```

Comment:
I think there is a small logic error here. In case of a team exercise, we would like to export the repository only, if the team was mentioned in the "studentList", otherwise we would export the same repository multiple times, if the students of the same team are mentioned here.

So I would suggest we rename studentList to participantList and instructors need to refer to team short names for a team exercise and to student logins for individual exercises.

If you don't want to change this here, please add a TODO and fix it in a subsequent PR

Reply:
You are correct. Right now the same repository would get exported multiple times if more than one student from the same team is entered.

Your suggestion for how to handle this makes total sense. I'll include it here since it should be quick to adapt.

Reply:
Export for individual-mode exercises:
![OnPaste 20200329-153729](https://user-images.githubusercontent.com/7109225/77850598-eb6c4180-71d3-11ea-87c1-91d32c330507.png)

Export for team-mode exercises:
![OnPaste 20200329-153907](https://user-images.githubusercontent.com/7109225/77850600-ec04d800-71d3-11ea-8d5c-6937163815ee.png)

I tested the export for individual and team exercises with a specified list of participants and without (i.e. download all).